File: src/main/java/com/example/game/units/Griffin.java
package com.example.game.units;

import com.example.game.core.Point2D;
import com.example.game.core.Unit;

public class Griffin extends Unit {
    public Griffin(int id, Point2D startPos) {
        super("그리핀", id, true, startPos);
    }

    @Override
    public void moveTo(Point2D dest) {
        log("날아서 " + position + " -> " + dest + " 로 이동합니다.");
        this.position = dest;
    }

    @Override
    protected boolean canAttack(Unit target) {
        // 그리핀은 "지상 유닛만 공격 가능"
        return !target.isFlying();
    }
}

File: src/main/kotlin/com/example/game/units/Griffin.kt
package com.example.game.units

import com.example.game.core.Point2D
import com.example.game.core.UnitBase
import com.example.game.traits.Attackable
import com.example.game.traits.Movable

/**
 * Griffin(그리핀): 하늘을 날아 이동하며, 번개로 지상 유닛만 공격한다.
 * 비행 유닛(Shuttle, Griffin)은 규칙상 공격 불가.
 */
class Griffin(id: Int, start: Point2D) :
    UnitBase(id = id, displayName = "그리핀", position = start, isFlying = true),
    Movable, Attackable {

    override fun moveTo(dest: Point2D) {
        log("하늘을 날아 $position → $dest 로 이동합니다.")
        position = dest
    }

    override fun attack(target: UnitBase) {
        if (target.isFlying) {
            log("공격 불가: 날아다니는 대상(${target.displayName}#${target.id})에게는 번개를 내릴 수 없습니다.")
            return
        }
        log("하늘에서 번개를 내려 ${target.displayName}#${target.id} 를 공격합니다!")
    }
}